### ISSUE TITLE
TransportServer CRD lacks HTTP (and, in general, L7) monitors

### ISSUE THREAD

**Issue by fabriziofiorucci** [F5Networks/k8s-bigip-ctlr#2110]
#### Title
TransportServer CRD lacks HTTP (and, in general, L7) monitors

#### Description
Let's assume an architecture where we have:

[client]---[LTM]---[NGINX KIC]---[Application pod]

LTM publishes a FastL4 virtual server. The need is:

- Use FastL4 VS on LTM
- LTM must run L7 (ie. HTTP/HTTPS) monitors towards NGINX KIC that it balances
- LTM configuration must be handled by CIS
- LTM must set its vs as available/not available based on L7 monitoring, even if the actual VS is FastL4

As of today the TransportServer CRD doesn't allow the use of http/https monitors. AS3 (and LTM itself) support defining L7 monitors at the pool level for L4 virtual servers.

The feature request is for something like:

```
apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  labels:
    f5cr: "true"
  name: xyz
spec:
  virtualServerAddress: 10.11.12.13
  virtualServerPort: 443
  virtualServerName: xyz
  mode: performance
  snat: auto
  pool:
    service: f5-nginx-ingress
    servicePort: 443
    monitor:
      type: http
      [monitor parameters]
```

#### Solution Proposed
As of today the TransportServer CRD doesn't allow the use of http/https monitors. This should be supported.

**Comment by mdditt2000** [comment]
@fabriziofiorucci Could you use IngressLink to accomplish [client]---[LTM]---[NGINX KIC] integration. IngressLink is coded to monitor the NGINX KIC liveness port. Here is a great writeup

https://github.com/mdditt2000/k8s-bigip-ctlr/blob/main/user_guides/ingresslink/clusterip/README.md

Please look at the video demo https://www.youtube.com/watch?v=Ieq9rzelpUk

This should resolve your issues simply. If you dont want to use IngressLink please let me know.

**Comment by fabriziofiorucci** [comment]
Thank you @mdditt2000 I will check this and let you know.

**Comment by myf5** [comment]
I think  is Readiness port? but this is not for real business status.

> @fabriziofiorucci Could you use IngressLink to accomplish [client]---[LTM]---[NGINX KIC] integration. IngressLink is coded to monitor the NGINX KIC liveness port. Here is a great writeup
> 
> https://github.com/mdditt2000/k8s-bigip-ctlr/blob/main/user_guides/ingresslink/clusterip/README.md
> 
> Please look at the video demo https://www.youtube.com/watch?v=Ieq9rzelpUk
> 
> This should resolve your issues simply. If you dont want to use IngressLink please let me know.

**Comment by myf5** [comment]
A possible way:
1. Expose business svc as usual, using NGINX Plus ingress controller
2. Expose  he NIC as a k8s svc,  then use ingress again(only for CIS) to expose the NGINX svc self.
In the CIS ingress, set:
```
HealthMonitorAnnotation: `[
     {
      "path":     "example.com/coffe",
      "send":     "HTTP GET /coffe",
      "interval": 5,
      "timeout":  5
     }, {
      "path":     "exmaple.com/tea",
      "send":     "HTTP GET /health/baz",
      "interval": 5,
      "timeout":  5
     }, {
      "path":     "svc1.bar.com/foo",
      "send":     "HTTP GET /health/foo",
      "interval": 5,
      "timeout":  10
     }
    ]`,
    }
```

**Comment by mdditt2000** [comment]
@myf5 i believe you are looking for multiple monitors, something like

```
 monitors:
        - type: https
          send: "GET /tea HTTP/1.1\r\nHost: cafe.example.com\r\n"
          recv: ""
          interval: 10
          timeout: 10
        - type: https
          send: "GET /coffee HTTP/1.1\r\nHost: cafe.example.com\r\n"
          recv: ""
          interval: 10
          timeout: 10
```

This is what we are adding to ExternalDNS CRDs. Would you like to see this in VS, TS CRDs.

**Comment by myf5** [comment]
VS, TS should include same ability.

**Comment by trinaths** [comment]
TCP health monitor support available from CIS 2.10. See examples https://github.com/F5Networks/k8s-bigip-ctlr/tree/master/docs/config_examples/customResource/VirtualServer/HealthMonitor
